File: electrician-management/src/main/java/com/electrician/electrician_management/service/WorkerService.java
package com.electrician.electrician_management.service;

import com.electrician.electrician_management.Entity.Worker;

import java.util.List;

public interface WorkerService {

    Worker getWorkerByPhone(String phone);

    Worker addWorker(Worker worker,Long electricianId);

    Worker getWorkerById(Long workerId);

    List<Worker> getWorkersByElectrician(Long electricianId);

    Worker updateWorker(Long workerId ,Worker worker);
    void deleteWorker(Long workerId);
}

File: electrician-frontend/src/app/core/services/worker.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class WorkerService {

  private baseUrl = environment.apiUrl + '/electrician/worker';

  constructor(private http: HttpClient) {}

  // ADD
  addWorker(worker: any, electricianId: number) {
    return this.http.post(
      `${this.baseUrl}/add/${electricianId}`,
      worker
    );
  }

  // LIST
  getWorkers(electricianId: number) {
    return this.http.get<any[]>(
      `${this.baseUrl}/list/${electricianId}`
    );
  }

  // ✅ GET SINGLE WORKER (🔥 REQUIRED FOR EDIT)
  getWorkerById(workerId: number) {
    return this.http.get<any>(
      `${this.baseUrl}/${workerId}`
    );
  }

  // UPDATE
  updateWorker(workerId: number, worker: any) {
    return this.http.put(
      `${this.baseUrl}/update/${workerId}`,
      worker
    );
  }

  // DELETE
  deleteWorker(workerId: number) {
    return this.http.delete(
      `${this.baseUrl}/delete/${workerId}`
    );
  }
}
